onnxruntime/test/onnx/main.cc

```diff
@@ -286,147 +287,160 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
-      {"GLU", "disable reason"},
-      {"GLU_dim", "disable reason"},
-      {"Linear", "disable reason"},
-      {"PReLU_1d", "disable reason"},
-      {"PReLU_1d_multiparam", "disable reason"},
-      {"PReLU_2d", "disable reason"},
-      {"PReLU_2d_multiparam", "disable reason"},
-      {"PReLU_3d", "disable reason"},
-      {"PReLU_3d_multiparam", "disable reason"},
-      {"PoissonNLLLLoss_no_reduce", "disable reason"},
-      {"Softsign", "disable reason"},
-      {"convtranspose_1d", "disable reason"},
-      {"convtranspose_3d", "disable reason"},
-      {"flatten_axis0", "disable reason"},
-      {"flatten_axis1", "disable reason"},
-      {"flatten_axis2", "disable reason"},
-      {"flatten_axis3", "disable reason"},
-      {"flatten_default_axis", "disable reason"},
-      {"gemm_broadcast", "disable reason"},
-      {"gemm_nobroadcast", "disable reason"},
-      {"matmul_2d", "disable reason"},
-      {"matmul_3d", "disable reason"},
-      {"matmul_4d", "disable reason"},
-      {"mvn", "disable reason"},
-      {"operator_add_broadcast", "disable reason"},
-      {"operator_add_size1_broadcast", "disable reason"},
-      {"operator_add_size1_right_broadcast", "disable reason"},
-      {"operator_add_size1_singleton_broadcast", "disable reason"},
-      {"operator_addconstant", "disable reason"},
-      {"operator_addmm", "disable reason"},
-      {"operator_basic", "disable reason"},
-      {"operator_lstm", "disable reason"},
-      {"operator_mm", "disable reason"},
-      {"operator_non_float_params", "disable reason"},
-      {"operator_params", "disable reason"},
-      {"operator_pow", "disable reason"},
-      {"operator_rnn", "disable reason"},
-      {"operator_rnn_single_layer", "disable reason"},
-      {"prelu_broadcast", "disable reason"},
-      {"prelu_example", "disable reason"},
-      {"cast_STRING_to_FLOAT", "Cast opset 9 not supported yet"},
-      {"cast_FLOAT_to_STRING", "Cast opset 9 not supported yet"},
-      {"tf_inception_resnet_v2", "Cast opset 9 not supported yet"},
-      {"tf_inception_v4", "Cast opset 9 not supported yet"},
-      {"tf_nasnet_large", "disable temporarily"},
-      {"tf_nasnet_mobile", "disable temporarily"},
-      {"tf_pnasnet_large", "disable temporarily"},
-      {"shrink", "test case is wrong"},
-      {"maxpool_2d_precomputed_strides", "ShapeInferenceError"},
-      {"averagepool_2d_precomputed_strides", "ShapeInferenceError"},
-      {"maxpool_with_argmax_2d_precomputed_strides", "ShapeInferenceError"},
-      {"tf_inception_v2", "result mismatch"},
-      {"tf_mobilenet_v2_1.0_224", "result mismatch"},
-      {"tf_mobilenet_v2_1.4_224", "result mismatch"},
-      {"tf_mobilenet_v1_1.0_224", "result mismatch"},
-      {"mobilenetv2-1.0", "result mismatch"},
-      {"mxnet_arcface", "result mismatch"}
+
+  struct BrokenTest
+  {
+    std::string test_name_;
+    std::set<int64_t> broken_versions_; // apply to all versions if empty
```

Comment:
>std::set<int64_t> broken_versions_; [](start = 4, length = 35)

Looks like mostly empty and a lot of typing for each entry. Perhaps, you can move as the last member and then omit empty init all together?

Reply:
Good idea.